### ISSUE TITLE
Questions about the equations in the paper

### ISSUE THREAD

**Issue by kaikai23** [MengLcool/AdaViT-pytorch#8]
Hey, nice and cool work! I want to consult about some implementation detials of some equations in the paper:

1. In the equation (8), the softmax is applied when the head mask is 1, and identity is applied when the head mask is 0. How do you actually implement this? Because during inference, the Q,K,V are all tensors of shape (N,h,197,d/h), and in a standard ViT, Q would be multiplied with K^T. If you only select a partial set of heads to compute, how do you implement this efficiently during inference?

2. In the equation (9), the concatenated tensor is multiplied with the linear weights W(superscript O', subscript l). When the resulting concatenated tensor has dynamic dimensions, how do you use a constant-sized  trainable parameter W? Or do you use other ways.

3. When you drop a patch / head, I wonder are they also dropped for all later layers, or are they only dropped for the current layer?

Looking forward to you reply, thank you!
